src/Tasks.UnitTests/ResourceHandling/GenerateResource_Tests.cs

```diff
@@ -477,7 +479,11 @@ public void ForceOutOfDateLinked(bool usePreserialized)
                 else
                 {
                     Utilities.AssertLogContainsResource(t, "GenerateResource.PreserializedResourcesRequiresExtensions");
+                    Utilities.AssertLogContainsResource(t, "GenerateResource.CorruptOutput", outputResource);
```

Comment:
Why does this output if we successfully delete?

Reply:
For example:

```
        ERROR : MSB3822: Non-string resources require the System.Resources.Extensions assembly at runtime, but it was not found in this project's references.
        WARNING MSB3555: Output file "C:\Users\raines\AppData\Local\Temp\rskgseqm.ymd\tmp363846e55b234157949d3501a7909105.resources" is possibly corrupt.
```

I don't love the idea of emitting a warning in this case in addition to the error. But I like using the existing method that does almost exactly what I want here.

Reply:
Now I have more questions. If the corrupt file is in temp, how does it break incremental build? We go ahead and move it to final location after failure?

Reply:
Or that's just the actual path/name for the test?

Reply:
Correct, that's verbatim from a test run of mine, which generates a `.resx` in temp and the `.resources` next to it. In a real build, it'd be in `obj` in a predictable location, and GenerateResources knows enough to predict that location and check its timestamp, then skip.

Reply:
Here's a real error situation:

```
S:\msbuild\artifacts\bin\bootstrap\net472\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(3056,5): error : MSB3822: Non-string resources require the System.Resources.Extensions assembly at runtime, but it was not found in this project's references. [S:\repro\microsoft\msbuild\issues\4553\BinaryResourceTest\BinaryResourceTest.csproj]
S:\msbuild\artifacts\bin\bootstrap\net472\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(3056,5): warning MSB3555: Output file "S:\repro\microsoft\msbuild\issues\4553\BinaryResourceTest\obj\Debug\net40\BinaryResourceTest.Resource1.resources" is possibly corrupt. [S:\repro\microsoft\msbuild\issues\4553\BinaryResourceTest\BinaryResourceTest.csproj]
```

Reply:
The warning is a bit spammy given that it will be deleted and if we failed to delete there will be more output. But I can live with this to keep the fix scoped at this late stage of the release.

Reply:
My thoughts exactly. Filed #4648.